server/auth/jwt_test.go

```diff
@@ -132,6 +132,35 @@ func testJWTInfo(t *testing.T, opts map[string]string) {
 	}
 }
 
+func TestJWTNoUsernameRevision(t *testing.T) {
+	optsMap := map[string]map[string]string{
+		"RSA": {
+			"pub-key":     jwtRSAPubKey,
+			"priv-key":    jwtRSAPrivKey,
+			"sign-method": "RS256",
+		},
+	}
+
+	t.Run("RSA", func(tt *testing.T) {
+		testJWTNoUsernameRevisionInfo(tt, optsMap["RSA"], "NA")
+	})
+}
+
+func testJWTNoUsernameRevisionInfo(t *testing.T, opts map[string]string, flag string) {
+	lg := zap.NewNop()
+	jwt, err := newTokenProviderJWT(lg, opts)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ctx := context.TODO()
+
+	ai, ok := jwt.info(ctx, "aaa", 120)
```

Comment:
`aaa` isn't a valid JWT token at all. You need to generate a valid JWT token, but without `username` or `revision` field.

Reply:
Please refer to https://github.com/etcd-io/etcd/blob/5327d91c20935a4324ace7aefa4a703c1e0d76a2/server/auth/jwt.go#L89-L105

For example to generate a JWT token without `revision`
```
	tk := jwt.NewWithClaims(t.signMethod,
		jwt.MapClaims{
			"username": username,
			"exp":      time.Now().Add(t.ttl).Unix(),
		})

	token, err := tk.SignedString(t.key)
	if err != nil {
		t.lg.Debug(
			"failed to sign a JWT token",
			zap.String("user-name", username),
			zap.Uint64("revision", revision),
			zap.Error(err),
		)
		return "", err
	}
```

Reply:
Generated token without `username` and `revision` and used that in the test function